### DISCUSSION TITLE
I hope to use this plug-in on my company's computer to communicate with Obsidian at home

### DISCUSSION THREAD

**Discussion by zsgcgrs** [coddingtonbear/obsidian-web#136] [Q&A]
I hope to use this plug-in on my company's computer to communicate with Obsidian at home. It doesn't seem to work when I try, and I can't connect.

**Comment by coddingtonbear** [comment]
Doing something like that is unfortunately going to be fairly complicated for you because you'll need to be aware of things like:

1. What IP can your home computer be reached on from work?
2. Does your home computer have a firewall, or is there a firewall in place on the router via which that computer accesses the network?
3. Does your work computer have an outbound firewall? 

It's probably pretty tricky to pull off, unfortunately.  What I'd recommend is focusing on getting some kind of SSH access into your computer at home from work; once you have that, it's very easy for you to use "port forwarding" to make it at least a little easier.

**Reply by zsgcgrs** [reply]
Thank you very much for your reply! The reason for this is that the files in obsidian are not encrypted, so I don't want to use obsidian on the company's computer, but I need to cut some fragments and other web pages into obsdian. I tried to communicate through domain name and IP but failed. There is no problem with the firewall. I want to know if there is a plug-in or browsing restriction that requires local access.

**Reply by coddingtonbear** [reply]
I understand the desire to keep your personal notes off of your company computer -- I do that, too. 

I can tell you that the plugin (especially as of the latest version) does not care where your Obsidian instance is running -- it'll present you with the standard Chrome permission dialog for you to use for granting permission to whatever host name.

It's very difficult to narrow down this kind of problem unless you're very familiar with things like routing and firewalls, and I'd be very suspicious that you _at least_ have a router between your home computer and the internet which will complicate matters.

Again, to keep your problem as simple as possible, I'd really recommend first trying to figure out how to establish an SSH tunnel between your work computer and your home one.  That's a much more common problem for folks, and you can definitely find instructions covering that on the internet.  Once you have an SSH tunnel, it's very easy to forward ports between your work and home computer so you can connect to obsidian via that tunnel.

**Reply by zsgcgrs** [reply]
Thanks again, I'll try a new way.

**Reply by zsgcgrs** [reply]
After I changed the IP address in the plug-in: 127.0.0.1 to the local IP: 10.0.0.5, the plug-in cannot connect to obsidian. It seems to only recognize 127.0.0.1. Can you verify it for me?

**Reply by coddingtonbear** [reply]
Oo -- you might've run into the problem I fixed here: https://github.com/coddingtonbear/obsidian-web/discussions/134#discussioncomment-7573016 -- if you didn't have luck the other day, it may work now.  Still: there are lots of factors to sort out so you may be running into a different issue.

**Reply by zsgcgrs** [reply]
What great news, it's working
